sigstore/sign.py

```diff
@@ -188,3 +212,62 @@ class SigningResult(BaseModel):
     """
     A record of the Rekor log entry for the signing operation.
     """
+
+    def _to_bundle(self) -> Bundle:
+        """
+        Creates a Sigstore bundle (as defined by Sigstore's protobuf specs)
+        from this `SigningResult`.
+        """
+
+        # TODO: Should we include our Fulcio intermediate in the chain?
+        chain = X509CertificateChain(
+            certificates=[X509Certificate(raw_bytes=self.cert_pem.encode())]
+        )
+
+        inclusion_proof: InclusionProof | None = None
+        if self.log_entry.inclusion_proof is not None:
+            inclusion_proof = InclusionProof(
+                log_index=self.log_entry.inclusion_proof.log_index,
+                root_hash=bytes.fromhex(self.log_entry.inclusion_proof.root_hash),
+                tree_size=self.log_entry.inclusion_proof.tree_size,
+                hashes=[
+                    bytes.fromhex(h) for h in self.log_entry.inclusion_proof.hashes
+                ],
+            )
+
+        tlog_entry = TransparencyLogEntry(
+            log_index=self.log_entry.log_index,
+            log_id=LogId(key_id=bytes.fromhex(self.log_entry.log_id)),
+            # TODO: Maybe leave this field out? It appears to be optional
+            # and it's just hardcoded for us, since it's the only kind of Rekor
+            # entry we support.
+            kind_version=KindVersion(kind="hashedrekord", version="0.0.1"),
```

Comment:
cc @kommendorkapten do you have opinions on this? `kind_version` isn't marked as mandatory and the information it provides is probably duplicated in the log entry body itself, so I wasn't sure if it makes sense to include it here.

Reply:
See also https://github.com/sigstore/protobuf-specs/issues/52

If I'm reading correctly we should keep it here because it may disappear from the log entry later.

Reply:
Makes sense! In that case, am I right in thinking that `kind_version` should probably be marked as mandatory in the protobuf definitions?

Reply:
Thanks for the link @znewman01 !

Yes, that should be mandatory in the spec. It's being added as of this: https://github.com/sigstore/protobuf-specs/pull/64/files